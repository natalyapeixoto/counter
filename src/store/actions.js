
export const incrementAsync = () =>{
    return {
        type:'INCREMENT'
    }
} 

export const increment = () => {
    return dispatch => {
        setTimeout(()=>{
            dispatch(incrementAsync())
        },2000)
    }
}

export const decrement = () =>{
    return {
        type:'DECREMENT'
    }
} 

export const reset = () =>{
    return {
        type:'RESET'
    }
} 