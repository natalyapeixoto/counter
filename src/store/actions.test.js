import {incrementAsync, decrement, reset, increment} from './actions'

jest.useFakeTimers()

describe('actions', () => {
    describe('incrementAsync', () => {
        it('should return object with type INCREMENT', () => {
            const action = {
                type: 'INCREMENT'
            }
            expect(incrementAsync()).toEqual(action)
        })
    })
    describe('increment', () => {
        it('should return a snapshot', () => {
            expect(increment()).toMatchSnapshot()
        })
        it('should return a function', () => {
            expect(typeof increment()).toEqual('function')
        })
        it('should call a function', () => {
            const dispatch = jest.fn()
            increment()(dispatch)

            jest.advanceTimersByTime(2000)

            expect(dispatch).toHaveBeenCalledTimes(1)
            expect(setTimeout).toHaveBeenLastCalledWith(expect.any(Function), 2000);
        })
    })
    describe('decrement', () => {
        it('should return an object with type DECREMENT', () => {
            const action = {
                type: 'DECREMENT'
            }
            
            expect(decrement()).toEqual(action)
        })
    })
    describe('reset', () => {
        it('should return an object with type RESET', () => {
            const action = {
                type: 'RESET'
            }

            expect(reset()).toEqual(action)
        })
    })
})