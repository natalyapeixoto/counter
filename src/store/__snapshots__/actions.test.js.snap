// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions increment should return a snapshot 1`] = `[Function]`;
